packages/orchestration/src/orchestration-client.ts

```diff
@@ -1,113 +1,136 @@
+import type { OrchestrationStreamChunkResponse } from './orchestration-stream-chunk-response.js';
 import type { HttpDestinationOrFetchOptions } from '@sap-cloud-sdk/connectivity';
 
+const logger = createLogger({ messageContext: 'orchestration-client' });
+
 /**
  * Get the orchestration client.
  */
 export class OrchestrationClient {
-  /**
-   * Creates an instance of the orchestration client.
-   * @param config - Orchestration module configuration. This can either be an `OrchestrationModuleConfig` object or a JSON string obtained from AI Launchpad.
-   * @param deploymentConfig - Deployment configuration.
-   * @param destination - The destination to use for the request.
-   */
   constructor(
     private config: OrchestrationModuleConfig | string,
     private deploymentConfig?: ResourceGroupConfig,
     private destination?: HttpDestinationOrFetchOptions
-  ) {}
+  ) {
+    try {
+      if (typeof config === 'string') {
+        JSON.parse(config);
+      }
+    } catch (error) {
+      throw new Error(`Could not parse JSON: ${error}`);
+    }
+  }
 
-  /**
-   * Creates a completion for the chat messages.
-   * @param prompt - Prompt configuration.
-   * @param requestConfig - Request configuration.
-   * @returns The completion result.
-   */
   async chatCompletion(
     prompt?: Prompt,
     requestConfig?: CustomRequestConfig
   ): Promise<OrchestrationResponse> {
+    const response = await this.executeRequest({
+      prompt,
+      requestConfig,
+      stream: false
+    });
+    return new OrchestrationResponse(response);
+  }
+
+  async stream(
+    prompt?: Prompt,
+    controller = new AbortController(),
+    options?: StreamOptions,
+    requestConfig?: CustomRequestConfig
+  ): Promise<OrchestrationStreamResponse<OrchestrationStreamChunkResponse>> {
+    if (typeof this.config === 'string' && options) {
+      logger.warn(
+        'Stream options are not supported when using a JSON module config.'
+      );
+    }
+
+    return this.createStreamResponse(
+      {
+        prompt,
+        requestConfig,
+        streamOptions: options
+      },
+      controller
+    );
+  }
+
+  private async executeRequest(options: RequestOptions): Promise<HttpResponse> {
+    const { prompt, requestConfig, stream, streamOptions } = options;
+
     const body =
       typeof this.config === 'string'
-        ? constructCompletionPostRequestFromJson(this.config, prompt)
-        : constructCompletionPostRequest(this.config, prompt);
+        ? constructCompletionPostRequestFromJsonModuleConfig(
+            JSON.parse(this.config),
+            prompt,
+            stream
+          )
+        : constructCompletionPostRequest(
+            this.config,
+            prompt,
+            stream,
+            streamOptions
+          );
 
     const deploymentId = await resolveDeploymentId({
       scenarioId: 'orchestration',
-      resourceGroup: this.deploymentConfig?.resourceGroup
+      ...(this.deploymentConfig || {})
```

Comment:
[pp] I would prefer using `??`, which sounds more correct to me, checking undefined or null instead of falsy. Same to other occurrences.